codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/smoketests/SmokeTestsRunnerGenerator.kt

```diff
@@ -108,12 +128,72 @@ class SmokeTestsRunnerGenerator(
         writer.withInlineBlock("#L {", "}", service) {
             if (testCase.vendorParams.isPresent) {
                 testCase.vendorParams.get().members.forEach { vendorParam ->
-                    if (vendorParam.key.value == "region") {
-                        writeInline("#L = ", vendorParam.key.value.toCamelCase())
-                        declareSection(SmokeTestRegionDefault)
-                        write("#L", vendorParam.value.format())
-                    } else {
-                        write("#L = #L", vendorParam.key.value.toCamelCase(), vendorParam.value.format())
+                    when (vendorParam.key.value) {
+                        "region" -> {
+                            writeInline("#L = ", vendorParam.key.value.toCamelCase())
+                            declareSection(SmokeTestRegionDefault)
+                            write("#L", vendorParam.value.format())
+                        }
+                        "sigv4aRegionSet" -> {
```

Comment:
correctness: smithy-kotlin should not know anything about AWS-specific config options. `region` was fine because we've treated it like a generic client config option, but is there any way to lift the handling of these AWS-specific vendor params out of smithy-kotlin?

I'm not sure there is, since this implementation heavily relies on sections, but something to consider. Can a single "vendor params" section be declared and the name of the vendor param (i.e. sigv4aRegionSet) be passed in context to aws-sdk-kotlin?